rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -290,11 +323,15 @@ def decode_transaction(
         # check if any eth transfer happened in the transaction, including in internal transactions
         events = self._maybe_decode_simple_transactions(transaction, tx_receipt)
         action_items: list[ActionItem] = []
+        counterparties = set()
 
         # decode transaction logs from the receipt
         for tx_log in tx_receipt.logs:
-            event, new_action_items = self.decode_by_address_rules(tx_log, transaction, events, tx_receipt.logs, action_items)  # noqa: E501
+            decoding_output = self.decode_by_address_rules(tx_log, transaction, events, tx_receipt.logs, action_items)  # noqa: E501
+            event, new_action_items, counterparty = decoding_output
```

Comment:
no need to do this. Why not just access them like `decoding_output.event`, ... ?